File: NumberGuessingGame.java
import java.util.Random;
import java.util.Scanner;

public class NumberGuessingGame {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();
        boolean playAgain;

        System.out.println("Welcome to the Number Guessing Game!");
        System.out.print("Please enter your name: ");
        String playerName = scanner.nextLine();  // Get the player's name

        System.out.println("Hello, " + playerName + "! Get ready to test your guessing skills.");
        System.out.println("Try to guess the number I'm thinking of.");
        System.out.println("Good luck, " + playerName + "!\n");

        do {
            int randomNumber = random.nextInt(100) + 1; // Generate random number between 1 and 100
            int guess;
            int attempts = 0;
            int maxAttempts = 10; // Set the maximum number of attempts
            boolean hasGuessedCorrectly = false;

            System.out.println("Guess a number between 1 and 100. You have " + maxAttempts + " attempts.");

            while (attempts < maxAttempts && !hasGuessedCorrectly) {
                System.out.print("Enter your guess: ");
                guess = scanner.nextInt();
                attempts++;

                if (guess == randomNumber) {
                    System.out.println("Congratulations! You guessed the correct number.");
                    hasGuessedCorrectly = true;
                } else if (guess < randomNumber) {
                    System.out.println("Your guess is too low. Try again.");
                } else {
                    System.out.println("Your guess is too high. Try again.");
                }
            }

            if (!hasGuessedCorrectly) {
                System.out.println("Sorry, you've used all your attempts. The correct number was " + randomNumber);
            }

            System.out.println("You took " + attempts + " attempts.");
            System.out.print("Do you want to play again? (yes/no): ");
            playAgain = scanner.next().equalsIgnoreCase("yes");

        } while (playAgain);

        System.out.println("Thank you for playing!");

        scanner.close();
    }
}

File: projects/number-guessing-game/number_guessing_game.py
"""
Project: Number Guessing Game
Author: Akhlaque Nabi
GitHub: i-Akhlaque
Description: 
    A simple Python game for Hacktoberfest 2025. 
    The computer picks a random number and the user has to guess it.
"""

import random

def main():
    print("Welcome to the Number Guessing Game!")
    number = random.randint(1, 100)
    attempts = 0

    while True:
        guess = int(input("Enter your guess (1-100): "))
        attempts += 1

        if guess < number:
            print("Too low! Try again.")
        elif guess > number:
            print("Too high! Try again.")
        else:
            print(f"Congratulations! You guessed the number {number} in {attempts} attempts.")
            break

if __name__ == "__main__":
    main()
